tests/wpt/metadata/html/semantics/scripting-1/the-script-element/module/dynamic-import/string-compilation-other-document.html.ini

```diff
@@ -1,16 +1,3 @@
 [string-compilation-other-document.html]
-  [inline event handlers triggered by JS should successfully import]
-    expected: FAIL
-
   [setTimeout should successfully import]
     expected: FAIL
```

Comment:
For `setTimeout` triggered by string code, we currently failed at finding its correct execution context so it will fetch a wrong link.

Corresponding spec link: https://html.spec.whatwg.org/multipage/#timer-initialisation-steps

I haven't digged into it but I think we need to pass base url of the timer and we might need to save it in `JsTimerTask`.

I wonder if it's good to be a follow-up issue, if not, I can try to fix it in this PR!

(Interestingly, only Firefox executes correctly on all the `setTimeout` tests for dynamic import module 😂)

Reply:
I did a bit of reading in Gecko:
* they get the active script reference when setTimeout is invoked (https://searchfox.org/mozilla-central/rev/89814940895946b48b4c04c702efd2c676ec8e7e/dom/base/nsGlobalWindowInner.cpp#5718)
* this calls JS_GetScriptedCallerPrivate, which retrieves the private value from the JSScript that is currently running
* when it's time to execute the timer, they compile it and set the private value of the new JSScript to point to the active script reference that invoked setTimeout (https://searchfox.org/mozilla-central/rev/89814940895946b48b4c04c702efd2c676ec8e7e/dom/base/nsGlobalWindowInner.cpp#5749-5756)

Reply:
Thanks for the investigation! Sounds like we can set the private reference of the script and then get it when setTimeout is invoked so that we can get a proper base_url 👀

Reply:
Filed #27260